web/packages/teleport/src/Integrations/status/AwsOidc/Tasks/Tasks.tsx

```diff
@@ -0,0 +1,162 @@
+
+import { useServerSidePagination } from 'teleport/components/hooks';
+import { FeatureBox } from 'teleport/components/Layout';
+import { AwsOidcHeader } from 'teleport/Integrations/status/AwsOidc/AwsOidcHeader';
+import { AwsOidcTitle } from 'teleport/Integrations/status/AwsOidc/AwsOidcTitle';
+import { getResourceType } from 'teleport/Integrations/status/AwsOidc/helpers';
+import { TaskState } from 'teleport/Integrations/status/AwsOidc/Tasks/constants';
+import { Task } from 'teleport/Integrations/status/AwsOidc/Tasks/Task';
+import { useAwsOidcStatus } from 'teleport/Integrations/status/AwsOidc/useAwsOidcStatus';
+import { integrationService, UserTask } from 'teleport/services/integrations';
+
+export function Tasks() {
+  const { integrationAttempt } = useAwsOidcStatus();
+  const { data: integration } = integrationAttempt;
+  const [showTask, setShowTask] = useState<UserTask>(undefined);
+
+  const serverSidePagination = useServerSidePagination<UserTask>({
+    pageSize: 20,
+    fetchFunc: async () => {
+      const { items, nextKey } =
+        await integrationService.fetchIntegrationUserTasksList(
+          integration.name,
+          TaskState.Open
+        );
+      return { agents: items, nextKey };
+    },
+    clusterId: '',
+    params: {},
+  });
+
+  useEffect(() => {
+    serverSidePagination.fetch();
+  }, [integration]);
+
+  if (integrationAttempt.status === 'processing') {
+    return <Indicator />;
+  }
+
+  if (serverSidePagination.attempt.status === 'processing') {
+    return <Indicator />;
+  }
+
+  if (integrationAttempt.status === 'error') {
+    return <Danger>{integrationAttempt.statusText}</Danger>;
+  }
+
+  if (serverSidePagination.attempt.status === 'failed') {
+    return <Danger>{serverSidePagination.attempt.statusText}</Danger>;
+  }
+
+  function close(resolved: boolean) {
+    if (resolved) {
+      serverSidePagination.modifyFetchedData(p => ({
+        ...p,
+        agents: p.agents.filter(t => t.name !== showTask.name),
+      }));
+    }
+    setShowTask(undefined);
+  }
+
+  return (
+    <Flex>
+      <Flex
+        flexDirection="column"
+        css={`
+          flex-grow: 1;
+        `}
+      >
+        {integration && (
+          <AwsOidcHeader integration={integration} tasks={true} />
+        )}
+        <FeatureBox
+          css={{ maxWidth: '1400px', paddingTop: '16px', gap: '30px' }}
+        >
+          {integration && (
+            <AwsOidcTitle integration={integration} tasks={true} />
+          )}
+          <Table<UserTask>
+            data={serverSidePagination.fetchedData?.agents || []}
+            row={{
+              onClick: row => {
+                if (showTask === undefined) {
+                  setShowTask(row);
+                }
+              },
+              getStyle: () => {
+                if (showTask === undefined) {
+                  return { cursor: 'pointer' };
+                }
+              },
+            }}
```

Comment:
I think we don't need this. There is already a button to view the item, so we don't need the same action repeated for the entire row.
It also makes accessibility slightly worse, for example, now you can't select the row content because it opens the preview.

Reply:
No, if it was intentionally designed that way, I’m fine with it.